test_outputs/query_execution/pub_module_level_const_missing.snap

```diff
@@ -3,6 +3,18 @@ source: src/query.rs
 expression: "&query_execution_results"
 ---
 {
+  "./test_crates/global_value_marked_deprecated/": [
+    {
+      "name": String("CONST_BECOMING_STATIC"),
+      "path": List([
+        String("global_value_marked_deprecated"),
+        String("CONST_BECOMING_STATIC"),
+      ]),
+      "span_begin_line": Uint64(9),
+      "span_filename": String("src/lib.rs"),
+      "visibility_limit": String("public"),
+    },
+  ],
```

Comment:
This is actually a false-positive in the `pub_module_level_const_missing` lint, as of Rust 1.83 which relaxed the rules on when & how `static` items may be used in `const` items. Previously, assigning a `static` to a `const` item was always a hard error. As of 1.83, certain `static` items can be assigned to `const` items so we can't prove breakage _merely_ by saying "this item used to be `const` but now is `static`."

Care to fix that lint? All you need to do is flip the `baseline` type check from `... on Constant` to `... on GlobalValue`

If you're also interested in writing new lints that flag the new breakage behavior made possible by Rust 1.83, I'd be happy to point you in the right direction there as well!

Reply:
> Care to fix that lint? All you need to do is flip the baseline type check from ... on Constant to ... on GlobalValue

Definitely

> If you're also interested in writing new lints that flag the new breakage behavior made possible by Rust 1.83, I'd be happy to point you in the right direction there as well!

Would like to know more about this, would you write it down in a new issue or?

Reply:
>Care to fix that lint? All you need to do is flip the baseline type check from ... on Constant to ... on GlobalValue

I think we want to flip `current` to be `GlobalValue` since now a `const` became `static` should be allowed?

Reply:
Yes, of course. Thanks for the catch!